cachito/workers/tasks/rubygems.py

```diff
@@ -16,3 +23,61 @@ def cleanup_rubygems_request(request_id):
         "username": get_rubygems_nexus_username(request_id),
     }
     nexus.execute_script("rubygems_cleanup", payload)
+
+
+def _get_config_file_for_given_package(
+    dependencies, bundle_dir, package_source_dir, rubygems_hosted_url
+):
+    """
+    Get Bundler config file.
+
+    Returns a Bundler config file with a mirror set for RubyGems dependencies pointing to
+    `rubygems_hosted_repo` URL. All GIT dependencies are configured to be replaced by local git
+     repos.
+
+    :param dependencies: an array of dependencies (dictionaries) with keys
+        "name": package name,
+        "path": an absolute path to a locally downloaded git repo,
+        "kind": dependency kind
+    :param bundle_dir: and absolute path to a bundle root directory
+    :param package_source_dir: the root directory of given package
+    :param rubygems_hosted_url: URL pointing to a request specific RubyGems hosted repo with
+     hardcoded user credentials
+    :return: dict with "content", "path" and "type" keys
+    """
+    base_config = dedent(
+        f"""
+        # Make sure gems aren’t installed system-wide but only for the given project
+        BUNDLE_PATH: "vendor/bundle"
+        # Sets mirror for all RubyGems sources
+        BUNDLE_MIRROR__ALL: "{rubygems_hosted_url}"
+        # Turn off the probing
+        BUNDLE_MIRROR__ALL__FALLBACK_TIMEOUT: "false"
+        # Install only ruby platform gems (=> gems with native extensions are compiled from source).
+        # All gems should be platform independent already, so why not keep it here.
+        BUNDLE_FORCE_RUBY_PLATFORM: "true"
+        BUNDLE_DEPLOYMENT: "true"
```

Comment:
Will `BUNDLE_DEPLOYMENT: "true"` be universally applicable for all use cases? Or should we leave it to the user to set this?

Reply:
It seems like this doesn't concern how image is built but only how cachito operated internally. So It should be transparent to the users if I'm not mistaken. The behavior of this option is described here: https://bundler.io/man/bundle-install.1.html#DEPLOYMENT-MODE